compiler/runtime.go

```diff
@@ -28,19 +28,23 @@ type Func interface {
 
 type Evaluator interface {
 	Type() semantic.Type
-	EvalString(scope Scope) (string, error)
-	EvalInt(scope Scope) (int64, error)
-	EvalUInt(scope Scope) (uint64, error)
-	EvalFloat(scope Scope) (float64, error)
-	EvalBool(scope Scope) (bool, error)
-	EvalTime(scope Scope) (values.Time, error)
+	EvalString(scope Scope) (values.Value, error)
```

Comment:
summarize this change:  the old `Eval<Type>`  functions were retrieving the value, converting it to a `go` type and then returning it.  Then this value was taken and converted back into a `values.Value`
the problem is that converting a null value causes a panic
so this sends that null further up the chain (edited) 
hidden inside a `values.value`
so it's actually not a big change, but just a large number of tiny refactor changes